src/v/redpanda/admin/api-doc/security.json

```diff
@@ -112,4 +145,293 @@
             }
         }
     }
+},
+"/v1/security/roles": {
+    "get": {
+        "tags": [
+            "Roles"
+        ],
+        "summary": "List roles",
+        "operationId": "list_roles",
+        "consumes": [
+            "application/json"
+        ],
+        "parameters": [
+            {
+                "name": "filter",
+                "description": "Return only roles whose names match this substring",
+                "in": "query",
+                "required": false,
+                "type": "string"
+            },
+            {
+                "name": "user",
+                "description": "Return only roles which are assigned to this user",
+                "in": "query",
+                "required": false,
+                "type": "string"
+            }
+        ],
+        "produces": ["application/json"],
+        "responses": {
+            "200": {
+                "description": "OK",
+                "schema": {
+                    "$ref": "#/definitions/roles_list"
+                }
+            }
+        }
+    },
+    "post": {
+        "tags": [
+            "Roles"
+        ],
+        "summary": "Create a role",
+        "operationId": "create_role",
+        "consumes": [
+            "application/json"
+        ],
+        "parameters": [
+            {
+                "name": "role",
+                "in": "body",
+                "required": true,
+                "type": "string"
```

Comment:
I think you need a schema here

Reply:
Yeah, I misunderstood how the "body" parameter works in openapi 2.

Reply:
done